diff --git a/configure b/configure
index adead92..0f0dfc6 100755
--- a/configure
+++ b/configure
@@ -1917,9 +1917,9 @@ echo "${ECHO_T}$ac_cv_c_compiler_gnu" >&6
 GCC=`test $ac_compiler_gnu = yes && echo yes`
 ac_test_CFLAGS=${CFLAGS+set}
 ac_save_CFLAGS=$CFLAGS
-CFLAGS="-g"
-echo "$as_me:1921: checking whether $CC accepts -g" >&5
-echo $ECHO_N "checking whether $CC accepts -g... $ECHO_C" >&6
+CFLAGS=""
+echo "$as_me:1921: checking whether $CC accepts" >&5
+echo $ECHO_N "checking whether $CC accepts... $ECHO_C" >&6
 if test "${ac_cv_prog_cc_g+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
@@ -1961,13 +1961,13 @@ if test "$ac_test_CFLAGS" = set; then
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS=
   else
-    CFLAGS="-g"
+    CFLAGS=
   fi
 else
   if test "$GCC" = yes; then
-    CFLAGS="-O2"
+    CFLAGS=
   else
     CFLAGS=
   fi
@@ -2834,9 +2834,9 @@ echo "${ECHO_T}$ac_cv_cxx_compiler_gnu" >&6
 GXX=`test $ac_compiler_gnu = yes && echo yes`
 ac_test_CXXFLAGS=${CXXFLAGS+set}
 ac_save_CXXFLAGS=$CXXFLAGS
-CXXFLAGS="-g"
-echo "$as_me:2838: checking whether $CXX accepts -g" >&5
-echo $ECHO_N "checking whether $CXX accepts -g... $ECHO_C" >&6
+CXXFLAGS=""
+echo "$as_me:2838: checking whether $CXX accepts" >&5
+echo $ECHO_N "checking whether $CXX accepts... $ECHO_C" >&6
 if test "${ac_cv_prog_cxx_g+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
@@ -2878,13 +2878,13 @@ if test "$ac_test_CXXFLAGS" = set; then
   CXXFLAGS=$ac_save_CXXFLAGS
 elif test $ac_cv_prog_cxx_g = yes; then
   if test "$GXX" = yes; then
-    CXXFLAGS="-g -O2"
+    CXXFLAGS=""
   else
-    CXXFLAGS="-g"
+    CXXFLAGS=""
   fi
 else
   if test "$GXX" = yes; then
-    CXXFLAGS="-O2"
+    CXXFLAGS=""
   else
     CXXFLAGS=
   fi
@@ -3914,7 +3914,7 @@ all:
 	@echo 'ac_maketemp="${MAKE}"'
 EOF
 # GNU make sometimes prints "make[1]: Entering...", which would confuse us.
-eval `${MAKE-make} -f conftest.make 2>/dev/null | grep temp=`
+${MAKE-make} -f conftest.make 2>/dev/null
 if test -n "$ac_maketemp"; then
   eval ac_cv_prog_make_${ac_make}_set=yes
 else
@@ -5904,12 +5904,12 @@ EOF
 fi
 
 if test X"$CC_G_OPT" = X"" ; then
-	CC_G_OPT='-g'
+	CC_G_OPT=''
 	test -n "$GCC" && test "x${ac_cv_prog_cc_g}" != xyes && CC_G_OPT=''
 fi
 
 if test X"$CXX_G_OPT" = X"" ; then
-	CXX_G_OPT='-g'
+	CXX_G_OPT=''
 	test -n "$GXX" && test "x${ac_cv_prog_cxx_g}" != xyes && CXX_G_OPT=''
 fi
 
@@ -6104,62 +6104,6 @@ echo "${ECHO_T}$cf_cv_shlib_version" >&6
 	# Some less-capable ports of gcc support only -fpic
 	CC_SHARED_OPTS=
 
-	cf_try_fPIC=no
-	if test "$GCC" = yes
-	then
-		cf_try_fPIC=yes
-	else
-		case $cf_cv_system_name in
-		(*linux*)	# e.g., PGI compiler
-			cf_try_fPIC=yes
-			;;
-		esac
-	fi
-
-	if test "$cf_try_fPIC" = yes
-	then
-		echo "$as_me:6121: checking which $CC option to use" >&5
-echo $ECHO_N "checking which $CC option to use... $ECHO_C" >&6
-		cf_save_CFLAGS="$CFLAGS"
-		for CC_SHARED_OPTS in -fPIC -fpic ''
-		do
-			CFLAGS="$cf_save_CFLAGS $CC_SHARED_OPTS"
-			cat >conftest.$ac_ext <<_ACEOF
-#line 6128 "configure"
-#include "confdefs.h"
-#include <stdio.h>
-int
-main (void)
-{
-int x = 1
-  ;
-  return 0;
-}
-_ACEOF
-rm -f conftest.$ac_objext
-if { (eval echo "$as_me:6140: \"$ac_compile\"") >&5
-  (eval $ac_compile) 2>&5
-  ac_status=$?
-  echo "$as_me:6143: \$? = $ac_status" >&5
-  (exit $ac_status); } &&
-         { ac_try='test -s conftest.$ac_objext'
-  { (eval echo "$as_me:6146: \"$ac_try\"") >&5
-  (eval $ac_try) 2>&5
-  ac_status=$?
-  echo "$as_me:6149: \$? = $ac_status" >&5
-  (exit $ac_status); }; }; then
-  break
-else
-  echo "$as_me: failed program was:" >&5
-cat conftest.$ac_ext >&5
-fi
-rm -f conftest.$ac_objext conftest.$ac_ext
-		done
-		echo "$as_me:6158: result: $CC_SHARED_OPTS" >&5
-echo "${ECHO_T}$CC_SHARED_OPTS" >&6
-		CFLAGS="$cf_save_CFLAGS"
-	fi
-
 	cf_cv_shlib_version_infix=no
 
 	case $cf_cv_system_name in
@@ -6310,7 +6254,7 @@ echo "${ECHO_T}$cf_cv_ldflags_search_paths_first" >&6
 		fi
 		# tested with IRIX 5.2 and 'cc'.
 		if test "$GCC" != yes; then
-			CC_SHARED_OPTS='-KPIC'
+			CC_SHARED_OPTS=''
 			MK_SHARED_LIB='${CC} ${LDFLAGS} ${CFLAGS} -shared -rdata_shared -soname `basename $@` -o $@'
 		else
 			MK_SHARED_LIB='${CC} ${LDFLAGS} ${CFLAGS} -shared -Wl,-soname,`basename $@` -o $@'
@@ -6370,7 +6314,7 @@ CF_EOF
 		if test "$cf_cv_enable_rpath" = yes ; then
 			EXTRA_LDFLAGS="${cf_ld_rpath_opt}\${RPATH_LIST} $EXTRA_LDFLAGS"
 		fi
-		CC_SHARED_OPTS="$CC_SHARED_OPTS -DPIC"
+		CC_SHARED_OPTS="$CC_SHARED_OPTS"
 
 	test "$cf_cv_shlib_version" = auto && cf_cv_shlib_version=rel
 	if test "$cf_cv_shlib_version" = rel; then
@@ -6382,12 +6326,12 @@ CF_EOF
 		MK_SHARED_LIB='${CC} ${LDFLAGS} ${CFLAGS} -shared -Wl,-Bshareable,-soname,'$cf_cv_shared_soname',-stats,-lc -o $@'
 		;;
 	(nto-qnx*|openbsd*|freebsd[12].*)
-		CC_SHARED_OPTS="$CC_SHARED_OPTS -DPIC"
+		CC_SHARED_OPTS="$CC_SHARED_OPTS"
 		MK_SHARED_LIB='${LD} ${LDFLAGS} -Bshareable -o $@'
 		test "$cf_cv_shlib_version" = auto && cf_cv_shlib_version=rel
 		;;
 	(dragonfly*|freebsd*)
-		CC_SHARED_OPTS="$CC_SHARED_OPTS -DPIC"
+		CC_SHARED_OPTS="$CC_SHARED_OPTS"
 		if test "$DFT_LWR_MODEL" = "shared" && test "$cf_cv_enable_rpath" = yes ; then
 			LOCAL_LDFLAGS="${cf_ld_rpath_opt}\$(LOCAL_LIBDIR)"
 			LOCAL_LDFLAGS2="${cf_ld_rpath_opt}\${RPATH_LIST} $LOCAL_LDFLAGS"
@@ -6404,7 +6348,7 @@ CF_EOF
 		MK_SHARED_LIB='${CC} ${LDFLAGS} ${CFLAGS} -shared -Wl,-soname,'$cf_cv_shared_soname',-stats,-lc -o $@'
 		;;
 	(netbsd*)
-		CC_SHARED_OPTS="$CC_SHARED_OPTS -DPIC"
+		CC_SHARED_OPTS="$CC_SHARED_OPTS"
 		if test "$DFT_LWR_MODEL" = "shared" && test "$cf_cv_enable_rpath" = yes ; then
 			LOCAL_LDFLAGS="${cf_ld_rpath_opt}\$(LOCAL_LIBDIR)"
 			LOCAL_LDFLAGS2="$LOCAL_LDFLAGS"
@@ -6449,7 +6393,7 @@ CF_EOF
 	(sco3.2v5*)  # also uw2* and UW7: hops 13-Apr-98
 		# tested with osr5.0.5
 		if test "$GCC" != yes; then
-			CC_SHARED_OPTS='-belf -KPIC'
+			CC_SHARED_OPTS='-belf'
 		fi
 		MK_SHARED_LIB='${LD} ${LDFLAGS} -dy -G -h `basename $@ .${REL_VERSION}`.${ABI_VERSION} -o $@'
 		if test "$cf_cv_enable_rpath" = yes ; then
@@ -6463,7 +6407,7 @@ CF_EOF
 	(sunos4*)
 		# tested with SunOS 4.1.1 and gcc 2.7.0
 		if test "$GCC" != yes; then
-			CC_SHARED_OPTS='-KPIC'
+			CC_SHARED_OPTS=''
 		fi
 		MK_SHARED_LIB='${LD} ${LDFLAGS} -assert pure-text -o $@'
 		test "$cf_cv_shlib_version" = auto && cf_cv_shlib_version=rel
@@ -6488,7 +6432,7 @@ CF_EOF
 
 		if test "$GCC" != yes; then
 			cf_save_CFLAGS="$CFLAGS"
-			for cf_shared_opts in -xcode=pic32 -xcode=pic13 -KPIC -Kpic -O
+			for cf_shared_opts in -xcode=pic32 -xcode=pic13 -O
 			do
 				CFLAGS="$cf_shared_opts $cf_save_CFLAGS"
 				cat >conftest.$ac_ext <<_ACEOF
@@ -6532,7 +6476,7 @@ rm -f conftest.$ac_objext conftest.$ac_ext
 	(sysv5uw7*|unix_sv*)
 		# tested with UnixWare 7.1.0 (gcc 2.95.2 and cc)
 		if test "$GCC" != yes; then
-			CC_SHARED_OPTS='-KPIC'
+			CC_SHARED_OPTS=''
 		fi
 		MK_SHARED_LIB='${LD} ${LDFLAGS} -d y -G -o $@'
 		;;
@@ -13736,7 +13680,7 @@ echo "$as_me:13735: result: $with_install_o" >&5
 echo "${ECHO_T}$with_install_o" >&6
 if test "x$with_install_o" = xyes
 then
-	INSTALL_OPT_O=`id root|sed -e 's/uid=[0-9]*(/ -o /' -e 's/gid=[0-9]*(/ -g /' -e 's/ [^=[:space:]][^=[:space:]]*=.*/ /' -e 's/)//g'`
+	INSTALL_OPT_O=`id root|sed -e 's/uid=[0-9]*(/ -o /' -e 's/gid=[0-9]*(/ /' -e 's/ [^=[:space:]][^=[:space:]]*=.*/ /' -e 's/)//g'`
 else
 	INSTALL_OPT_O=
 fi
@@ -14620,7 +14564,7 @@ if test -n "$GCC"
 then
 	if test "$with_assertions" = no
 	then
-		CPPFLAGS="$CPPFLAGS -DNDEBUG"
+		CPPFLAGS="$CPPFLAGS"
 	else
 
  	ADAFLAGS="$ADAFLAGS -gnata"
@@ -14641,7 +14585,7 @@ cat >>confdefs.h <<EOF
 #define USE_DMALLOC 1
 EOF
 
-	: ${with_cflags:=-g}
+	: ${with_cflags:=}
 	: ${with_no_leaks:=yes}
 	 with_dmalloc=yes
 else
diff --git a/configure.in b/configure.in
index 7dbf48f..55b0fdc 100644
--- a/configure.in
+++ b/configure.in
@@ -1504,7 +1504,7 @@ if test -n "$GCC"
 then
 	if test "$with_assertions" = no
 	then
-		CPPFLAGS="$CPPFLAGS -DNDEBUG"
+		CPPFLAGS="$CPPFLAGS"
 	else
 		CF_ADD_ADAFLAGS(-gnata)
 	fi
diff --git a/include/MKterm.h.awk.in b/include/MKterm.h.awk.in
index c6dbccc..90049bb 100644
--- a/include/MKterm.h.awk.in
+++ b/include/MKterm.h.awk.in
@@ -126,7 +126,7 @@ BEGIN {
 	print  ""
 	print  "#else /* !HAVE_TERMIO_H */"
 	print  ""
-	print  "#if __MINGW32__"
+	print  "#if defined(__MINGW32__) || defined(_MSC_VER)"
 	print  "#  include <ncurses_mingw.h>"
 	print  "#  define TTY struct termios"
 	print  "#else"
diff --git a/include/nc_mingw.h b/include/nc_mingw.h
index ab30c09..97cc945 100644
--- a/include/nc_mingw.h
+++ b/include/nc_mingw.h
@@ -27,7 +27,7 @@
  ****************************************************************************/
 
 /****************************************************************************
- * Author: Thomas Dickey, 2008-on                                           * 
+ * Author: Thomas Dickey, 2008-on                                           *
  ****************************************************************************/
 
 /* $Id: nc_mingw.h,v 1.4 2017/07/22 17:06:38 tom Exp $ */
@@ -35,11 +35,11 @@
 #ifndef NC_MINGW_H
 #define NC_MINGW_H 1
 
-#ifdef __MINGW32__
+#if defined(__MINGW32__) || defined(_MSC_VER)
 #ifdef WINVER
 #  if WINVER < 0x0501
 #    error WINVER must at least be 0x0501
-#  endif  
+#  endif
 #else
 #  define WINVER 0x0501
 #endif
@@ -51,7 +51,9 @@
 #undef gettimeofday
 #define gettimeofday(tv,tz) _nc_gettimeofday(tv,tz)
 
+#if HAVE_SYS_TIME_H
 #include <sys/time.h>	/* for struct timeval */
+#endif
 
 extern int _nc_gettimeofday(struct timeval *, void *);
 
diff --git a/include/nc_termios.h b/include/nc_termios.h
index 5efab40..860d604 100644
--- a/include/nc_termios.h
+++ b/include/nc_termios.h
@@ -71,7 +71,7 @@
 
 #else /* !HAVE_TERMIO_H */
 
-#if __MINGW32__
+#if defined(__MINGW32__) || defined(_MSC_VER)
 
 /* lflag bits */
 #define ISIG	0x0001
diff --git a/include/ncurses_mingw.h b/include/ncurses_mingw.h
index 52473b2..2c0ac4b 100644
--- a/include/ncurses_mingw.h
+++ b/include/ncurses_mingw.h
@@ -27,7 +27,7 @@
  ****************************************************************************/
 
 /****************************************************************************
- * Author: Juergen Pfeifer, 2008-on                                         * 
+ * Author: Juergen Pfeifer, 2008-on                                         *
  ****************************************************************************/
 
 /* $Id: ncurses_mingw.h,v 1.4 2017/07/22 17:06:49 tom Exp $ */
@@ -37,7 +37,7 @@
  * to support I/O to external terminals with ncurses on the Windows OS.
  */
 
-#ifdef __MINGW32__
+#if defined(__MINGW32__) || defined(_MSC_VER)
 #ifndef _NC_MINGWH
 #define _NC_MINGWH
 
@@ -66,14 +66,14 @@ struct termios
 };
 
 extern NCURSES_EXPORT(int)  _nc_mingw_tcsetattr(
-    int fd, 
-    int optional_actions, 
+    int fd,
+    int optional_actions,
     const struct termios* arg);
 extern NCURSES_EXPORT(int)  _nc_mingw_tcgetattr(
-    int fd, 
+    int fd,
     struct termios* arg);
 extern NCURSES_EXPORT(int)  _nc_mingw_tcflush(
-    int fd, 
+    int fd,
     int queue);
 extern NCURSES_EXPORT(void) _nc_set_term_driver(void* term);
 
diff --git a/ncurses/win32con/gettimeofday.c b/ncurses/win32con/gettimeofday.c
index 867041f..3087f00 100644
--- a/ncurses/win32con/gettimeofday.c
+++ b/ncurses/win32con/gettimeofday.c
@@ -34,6 +34,7 @@
 #include <curses.priv.h>
 
 #include <windows.h>
+#include <winsock2.h>
 
 MODULE_ID("$Id: gettimeofday.c,v 1.3 2014/04/26 19:41:34 juergen Exp $")
 
diff --git a/ncurses/win32con/win_driver.c b/ncurses/win32con/win_driver.c
index 2a6ed00..75d9b55 100644
--- a/ncurses/win32con/win_driver.c
+++ b/ncurses/win32con/win_driver.c
@@ -56,7 +56,7 @@
 
 MODULE_ID("$Id: win_driver.c,v 1.59 2017/07/22 21:10:28 tom Exp $")
 
-#ifndef __GNUC__
+#if 0
 #  error We need GCC to compile for MinGW
 #endif
 
@@ -261,7 +261,7 @@ static BOOL
 con_write16(TERMINAL_CONTROL_BLOCK * TCB, int y, int x, cchar_t *str, int limit)
 {
     int actual = 0;
-    CHAR_INFO ci[limit];
+    CHAR_INFO * ci = (CHAR_INFO*) _alloca(sizeof(CHAR_INFO) * limit);
     COORD loc, siz;
     SMALL_RECT rec;
     int i;
@@ -310,7 +310,7 @@ con_write16(TERMINAL_CONTROL_BLOCK * TCB, int y, int x, cchar_t *str, int limit)
 static BOOL
 con_write8(TERMINAL_CONTROL_BLOCK * TCB, int y, int x, chtype *str, int n)
 {
-    CHAR_INFO ci[n];
+    CHAR_INFO * ci = (CHAR_INFO*) _alloca(sizeof(CHAR_INFO) * n);
     COORD loc, siz;
     SMALL_RECT rec;
     int i;
@@ -509,7 +509,7 @@ wcon_doupdate(TERMINAL_CONTROL_BLOCK * TCB)
 	if ((CurScreen(sp)->_clear || NewScreen(sp)->_clear)) {
 	    int x;
 #if USE_WIDEC_SUPPORT
-	    cchar_t empty[Width];
+	    cchar_t * empty = (cchar_t*) _alloca(sizeof(cchar_t) * Width);
 	    wchar_t blank[2] =
 	    {
 		L' ', L'\0'
@@ -518,7 +518,7 @@ wcon_doupdate(TERMINAL_CONTROL_BLOCK * TCB)
 	    for (x = 0; x < Width; x++)
 		setcchar(&empty[x], blank, 0, 0, 0);
 #else
-	    chtype empty[Width];
+	    chtype * empty = (chtype*) _alloca(sizeof(chtype) * Width);
 
 	    for (x = 0; x < Width; x++)
 		empty[x] = ' ';
@@ -674,8 +674,8 @@ wcon_dobeepflash(TERMINAL_CONTROL_BLOCK * TCB,
     int max_cells = (high * wide);
     int i;
 
-    CHAR_INFO this_screen[max_cells];
-    CHAR_INFO that_screen[max_cells];
+    CHAR_INFO * this_screen = (CHAR_INFO*) _alloca(sizeof(CHAR_INFO) * max_cells);
+    CHAR_INFO * that_screen = (CHAR_INFO*) _alloca(sizeof(CHAR_INFO) * max_cells);
     COORD this_size;
     SMALL_RECT this_region;
     COORD bufferCoord;
